features/430-passwordless/design.md

```diff
@@ -0,0 +1,204 @@
+model, there will be extra steps in the authentication session:
+1. Select Identity
+2. Perform Primary Authentication
+    - If multiple primary authenticators is applicable for the selected
+      identity, user is prompted to select one.
+3. Perform Secondary Authentication (if configured)
+    - If multiple secondary authenticators is set up for the user, user is
+      prompted to select one.
+    - If no secondary authenticators is configured:
+        - If secondary authentication is required, user is prompted to
+          setup a secondary authenticator.
+        - If secondary authentication is optional, this step is skipped.
+
+For examples:
+- Simple password login:
+    1. Enter Login ID.
+    2. Authenticate with password.
+    3. Done.
+- Password/SSO with email/TOTP MFA:
+    1. Select Login ID or OAuth identity.
+        - Login ID: enter email.
+        - OAuth: redirect to OAuth provider and wait for return.
+    2. Authenticate selected identity:
+        - Login ID: authenticate with password.
+        - OAuth: aftering returning, already authenticated by OAuth provider.
+    3. Select secondary authentication method (Email OTP/TOTP)
+        - Email OTP: send email with OTP and verify entered OTP.
+        - TOTP: verify entered TOTP.
+    4. Done.
+- WebAuthn password-less:
+    1. Enter Login ID.
+    2. Authenticate with security key through WebAuthn.
+    3. Done.
+- WebAuthn resident key authentication with email/password fallback:
+    1. Select Login ID or WebAuthn identity.
+        - Login ID: enter email.
+        - WebAuthn: request WebAuthn with resident key.
+    2. Authenticate selected identity.
+        - Login ID: select authenticator.
+            - Password: authenticate with password.
+            - WebAuthn: authenticate with security key.
+        - WebAuthn: already authenticated using resident key.
+    3. Done.
+- Email OTP passwordless/SSO:
+    1. Select Login ID or OAuth identity.
+        - Login ID: enter email.
+        - OAuth: redirect to OAuth provider and wait for return.
+    2. Authenticate selected identity:
+        - Login ID: send email with OTP and verify entered OTP.
+        - OAuth: aftering returning, already authenticated by OAuth provider.
+    3. Done.
+- GitHub:
+    1. Enter Login ID.
+    2. Authenticate with password.
+    3. Select secondary authentication method (WebAuthn/Recovery Code/Email OTP/TOTP)
+        - Omitted for brevity.
+    4. Done.
+
+## OIDC claims
+
+To indicate authenticator used in authentication, `amr` claim is used in OIDC
+ID token.
+
+`amr` claim is an array of string. It includes authentication method used:
+- If password authenticator is used: `pwd` is included.
+- If any OTP (TOTP/OOB-OTP) is used: `otp` is included.
+- If WebAuthn is used: `hwk` is included.
+
+If no authentication method is to be included in `amr` claim, `amr` claim would
+be omitted from the ID token.
```

Comment:
Would like to know which case the id token will not have the amr?

Reply:
When user is logged in with OAuth only, there will be no `amr` claim.